### PR TITLE
[SPARK-42284][CONNECT] Make sure connect server assembly is built before running client tests - SBT

### PR DESCRIPTION
**Pull request by hvanhovell**
### What changes were proposed in this pull request?
This PR makes sure SBT assembles the connect server assembly jar before we run client tests.

### Why are the changes needed?
It makes it easier to develop and test the JVM client for Spark Connect.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Manually tested.

### CODE REVIEW

**Comment by hvanhovell** [comment]
cc @zhenlineo @LuciferYang

File: project/SparkBuild.scala [hunk 1]
```diff
@@ -851,6 +851,9 @@ object SparkConnectClient {
       )
     },
 
+    // Make sure the connect server assembly jar is available for testing.
+    test := ((Test / test) dependsOn (LocalProject("connect") / assembly)).value,
```
Comment (LuciferYang) [inline project/SparkBuild.scala:855]: looks perfect,  does maven have a similar solution?

**Review by LuciferYang — APPROVED** [review]
Manual check

```
gh pr checkout 39854
build/sbt clean
build/sbt "connect-client-jvm/test"
```

All test passed. 

LGTM & Thanks @hvanhovell

File: project/SparkBuild.scala (hunk 1 above)
Comment (LuciferYang) [reply project/SparkBuild.scala:855]: > does maven have a similar solution？

Let me investigate

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.4.

**Pull request closed without merging**